### ISSUE TITLE
jpaQueryFactory.update generate wrong sql in spring boot 2.2.2

### ISSUE THREAD

**Issue by zqr95518** [querydsl/querydsl#2511]
After update project to spring boot 2.2.2. Querydsl 4.2.1.

**Comment by zqr95518** [comment]
```
 jpaQueryFactory.update(induction).set(induction.autoPublish, enable).where(induction.base.id.eq(baseId)).execute();
```
Will generate where id = baseId not base.id = baseId.
No problem with spring boot 2.1.10 .

**Comment by x403368945** [comment]
Serious bugs!!! Serious bugs!!! Serious bugs!!!

// ****************************
// Versions
// SpringBoot: 2.1.5.RELEASE
// Hibernate: 5.3.10.Final
// QueryDSL:  4.2.1
// ****************************
// Update Expression
<pre>
new JPAQueryFactory(entityManager)
        .update(QTabTest.tabTest)
        .set(QTabTest.tabTest.cipherText, "AAA")
        .set(QTabTest.tabTest.plainText, "AAA")
        .where(q.id.eq(1L))
        .execute()
        ;
</pre>
// Compile: JPQLSerializer.java#visitConstant  -> Line Numbers: 325-331
<pre>
    update TabTest tabTest
    set tabTest.cipherText=?1,tabTest.plainText=?1
    where tabTest.id = ?2
</pre>
// Execute SQL:
<pre>
// Expect SQL
update tab_test
    set cipher_text='e1faffb3e614e6c2fba74296962386b7',
    plain_text='AAA'
    where id = 1;
// Real SQL
update tab_test
    set cipher_text='e1faffb3e614e6c2fba74296962386b7',
    plain_text='e1faffb3e614e6c2fba74296962386b7'
    where id = 1;
</pre>
// Entity
<pre>
@Table(name = "tab_test")
@Entity
@Data
public class TabTest {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Convert(converter = MD5Converter.class) // Convert ****************************************************
    private String cipherText;
    private String plainText;
}
</pre>
// Convert
<pre>
public class MD5Converter implements AttributeConverter<String, String> {
    @Override
    public String convertToDatabaseColumn(final String attribute) {
        return DigestUtils.md5Hex(attribute);
    }

    @Override
    public String convertToEntityAttribute(final String dbData) {
        return dbData;
    }
}
</pre>

**Comment by idosal** [comment]
Thanks for reporting these issues.
In the next few days I'll introduce compatibility fixes involving Hibernate 5.2+ that probably touch the root causes, though I'm not sure they'd fix them. If you're in a rush, you can try testing with this [PR](https://github.com/querydsl/querydsl/pull/2354) applied.

**Comment by jwgmeligmeyling** [comment]
Between 4.2.0 / 4.1.4 and 4.2.1 JPA was barely touched, let alone the `JPAUpdateClause`. 

Please specify the last version in which this worked.
